### PR TITLE
[CARBONDATA-3668] CarbonSession should use old flow (non-CarbonExtensions flow)

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
 Considering back-compatibility,  CarbonSession should use old flow (not CarbonExtensions flow)

 ### What changes were proposed in this PR?
1. Remove CarbonExtension from CarbonSession
2. Recover CarbonSessionCatalog
 ### Does this PR introduce any user interface change?
 - No
 - Yes. (please explain the change and update document)

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala [hunk 1]
```diff
@@ -50,7 +48,7 @@ object CarbonThriftServer {
     val builder = SparkSession
```
Comment (zzcclp) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: the 'storePath' parameter is not required, can remove it.

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1702/

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala (hunk 1 above)
Comment (QiangCai) [reply integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]: fixed

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1709/

**Comment by zzcclp** [comment]
please rebase master

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/5/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/144/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1848/

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala [hunk 2]
```diff
@@ -28,70 +28,63 @@ import org.apache.carbondata.common.logging.LogServiceFactory
...
 import org.apache.carbondata.spark.util.CarbonSparkUtil
 
- /**
-  * CarbonThriftServer support different modes:
-  * 1. read/write data from/to HDFS or local,it only needs configurate storePath
-  * 2. read/write data from/to S3, it needs provide access-key, secret-key, s3-endpoint
-  */
+/**
+ * CarbonThriftServer support different modes:
+ * 1. read/write data from/to HDFS or local, no need parameters
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/thriftserver/CarbonThriftServer.scala]:
```suggestion
 * 1. read/write data from/to HDFS or local, no parameter is required in this case
```

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSessionExample.scala [hunk 3]
```diff
@@ -40,11 +40,18 @@ object CarbonSessionExample {
       .addProperty(CarbonCommonConstants.ENABLE_QUERY_STATISTICS, "false")
     val spark = ExampleUtils.createCarbonSession("CarbonSessionExample")
     spark.sparkContext.setLogLevel("error")
-    exampleBody(spark)
+    Seq(
+      "stored as carbondata",
+      "using carbondata",
+      "stored by 'carbondata'",
```
Comment (jackylk) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSessionExample.scala]: Is it still supported?

Comment (QiangCai) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSessionExample.scala]: yes, CarbonSession will keep all old features.

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.4/149/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1853/

**Comment by jackylk** [comment]
LGTM

**Comment by QiangCai** [comment]
I will raise another PR to fix the ci issue for spark 2.4

**Pull request closed without merging**